dataconf/utils.py

```diff
@@ -124,6 +124,27 @@ def __parse(value: any, clazz: Type, path: str, strict: bool, ignore_unexpected:
             ]
         return None
 
+    if origin is tuple:
+        if value is not None:
+            if len(args) < 1:
+                raise MissingTypeException(
+                    "expected tuple with type information: Tuple[?]"
+                )
+            has_ellipsis = args[-1] == Ellipsis
+            if has_ellipsis and len(args) != 2:
+                raise MissingTypeException(
+                    "expected one type since ellipsis is used: Tuple[?, ...]"
+                )
+            _args = args if not has_ellipsis else [args[0]] * len(value)
+            if len(value) > 0 and len(value) != len(_args):
+                raise MalformedConfigException(
+                    "number of provided values does not match expected number of values for tuple."
+                )
+            return tuple(
+                __parse(v, arg, f"{path}[]", strict, ignore_unexpected)
+                for v, arg in zip(value, _args)
+            )
+
```

Comment:
I missed that case, shall it be handle explicitly or even throw? a null tuple? if not wrapped in an optional?

Reply:
I'm sorry, I don't think I understand what case you're talking about.

Reply:
what if `{"a": null}` and `tuple[int]` without default is expected? can you add a small test for that?

Reply:
I would rather make the parsing crash in that case (or delegate the value resolving to the nested __parse) because expecting a tuple and getting a null seems to break the safety this library wants to bring. One more little push and we are good to merge :)

Reply:
Sounds good to me. I have tweaked the logic for list and tuple parsing to make them raise a MalformedConfigException if the value is `None` and updated the tests to reflect that change. I'm not convinced that I like the messages that I put for the exceptions though. If you want to change those to something else, feel free to change them yourself or let me know what you want them to be and I can change them.